receiver/awscontainerinsightreceiver/internal/neuron/neuron_monitor_scraper.go

```diff
@@ -0,0 +1,208 @@
+	if opts.Consumer == nil {
+		return nil, errors.New("consumer cannot be nil")
+	}
+	if opts.Host == nil {
+		return nil, errors.New("host cannot be nil")
+	}
+	if opts.HostInfoProvider == nil {
+		return nil, errors.New("cluster name provider cannot be nil")
+	}
+
+	scrapeConfig := &config.ScrapeConfig{
+		ScrapeInterval: model.Duration(collectionInterval),
+		ScrapeTimeout:  model.Duration(collectionInterval),
+		JobName:        jobName,
+		Scheme:         "http",
+		MetricsPath:    "/metrics",
+		ServiceDiscoveryConfigs: discovery.Configs{
+			&kubernetes.SDConfig{
+				Role: kubernetes.RoleService,
+				NamespaceDiscovery: kubernetes.NamespaceDiscovery{
+					IncludeOwnNamespace: true,
+				},
+				Selectors: []kubernetes.SelectorConfig{
+					{
+						Role:  kubernetes.RoleService,
+						Label: "k8s-app=neuron-monitor-service",
+					},
+				},
+				AttachMetadata: kubernetes.AttachMetadataConfig{
+					Node: true,
+				},
+			},
+		},
+		RelabelConfigs: []*relabel.Config{
+			{
+				SourceLabels: model.LabelNames{"__address__"},
+				Regex:        relabel.MustNewRegexp("([^:]+)(?::\\d+)?"),
+				Replacement:  "${1}:8000",
+				TargetLabel:  "__address__",
+				Action:       relabel.Replace,
+			},
+		},
+		MetricRelabelConfigs: []*relabel.Config{
+			{
+				SourceLabels: model.LabelNames{"__name__"},
+				Regex:        relabel.MustNewRegexp("neuron.*|system_.*|execution_.*"),
+				Action:       relabel.Keep,
+			},
+			{
+				SourceLabels: model.LabelNames{"instance_name"},
+				TargetLabel:  "NodeName",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+			{
+				SourceLabels: model.LabelNames{"instance_id"},
+				TargetLabel:  "InstanceId",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+			{
+				SourceLabels: model.LabelNames{"neuroncore"},
+				TargetLabel:  "DeviceId",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+			// hacky way to inject static values (clusterName) to label set without additional processor
+			// relabel looks up an existing label then creates another label with given key (TargetLabel) and value (static)
+			{
+				SourceLabels: model.LabelNames{"instance_id"},
+				TargetLabel:  "ClusterName",
+				Regex:        relabel.MustNewRegexp("neuron.*"),
+				Replacement:  opts.HostInfoProvider.GetClusterName(),
+				Action:       relabel.Replace,
+			},
+			// hacky way to inject static values (namespace, podName, containerName) to label set without additional processor,
+			// we need these labels only when we have neuronCore label specified in logs
+			// we process and replace with actual podName, namespace and ContainerName
+			// relabel looks up an existing label then creates another label with given key (TargetLabel) and value (static)
```

Comment:
For now, just add a line here explaining why the placeholder value is the neuroncore (another component will replace it later).

Reply:
removed those injections we will add them manually in agent code